pkg/controllers/migration/virtualmachine.go

```diff
@@ -809,10 +785,12 @@ func (h *virtualMachineHandler) checkAndCreateVirtualMachineImage(vm *migration.
 	}
 
 	if vm.Spec.StorageClass != "" {
```

Comment:
we need more checks here to identify if storage class is using the longhorn v1 provisioner. It is possible to have multiple longhorn v1 storage classes where the replica counts may be different.